coldfront/core/project/views.py

```diff
@@ -3585,3 +3588,109 @@ def get_success_url(self):
         return reverse(
             'vector-project-request-detail',
             kwargs={'pk': self.kwargs.get('pk')})
+
+
+class SavioProjectUndenyRequestView(LoginRequiredMixin, UserPassesTestMixin, View):
+    login_url = '/'
+
+    def test_func(self):
+        """UserPassesTestMixin tests."""
+        if self.request.user.is_superuser:
+            return True
+
+        message = (
+            'You do not have permission to undeny a project request.')
+        messages.error(self.request, message)
+
+    def dispatch(self, request, *args, **kwargs):
+        project_request = get_object_or_404(
+            SavioProjectAllocationRequest, pk=self.kwargs.get('pk'))
+
+        state_status = savio_request_state_status(project_request)
+        denied_status = ProjectAllocationRequestStatusChoice.objects.get(name='Denied')
+
+        if state_status != denied_status:
+            message = 'Savio project request has an unexpected status.'
+            messages.error(request, message)
+
+            return HttpResponseRedirect(
+                reverse('savio-project-request-detail',
+                        kwargs={'pk': self.kwargs.get('pk')}))
+
+        return super().dispatch(request, *args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        project_request = get_object_or_404(
+            SavioProjectAllocationRequest, pk=kwargs.get('pk'))
+        
+        if project_request.state['eligibility']['status'] == 'Denied':
+            project_request.state['eligibility']['status'] = 'Pending'
+
+        if project_request.state['readiness']['status'] == 'Denied':
+            project_request.state['readiness']['status'] = 'Pending'
+            
+        if project_request.state['other']['timestamp']:
+            project_request.state['other']['justification'] = ''
+            project_request.state['other']['timestamp'] = ''
+
+        project_request.status = ProjectAllocationRequestStatusChoice.objects.get(name="Under Review")
```

Comment:
Setting this to `savio_request_state_status(project_request)` would be more appropriate. It could've been in the `Approved - Processing` state when it was denied; if so, it should revert to that.